fuel-vm/src/error.rs

```diff
@@ -367,6 +367,16 @@ pub enum BugVariant {
     /// Receipts context is full, but there's an attempt to add more receipts.
     #[strum(message = "Receipts context is full, cannot add new receipts.")]
     ReceiptsCtxFull,
+
+    /// Witness index is out of bounds.
+    #[strum(message = "Witness index is out of bounds.")]
+    WitnessIndexOutOfBounds,
+
+    /// The witness part index is higher than the total number of parts.
+    #[strum(
+        message = "The witness part index is higher than the total number of parts."
+    )]
+    NextPartIndexIsHigherThanTotalNumberOfParts,
```

Comment:
I know this was already decided and there are a lot of places that use this term (this applies to any place using the term `Part`), but I think I'd prefer something like `Subsection` over the term `Part`. `Part` is really vague, and `Subsection` is a bit more clear IMO.

Reply:
Subsection sounds better, I will update it. I will start with https://github.com/FuelLabs/fuel-vm/pull/714 first.

Reply:
https://github.com/FuelLabs/fuel-vm/pull/714/commits/b4f2cdd65888a98ee045eefe9ecd4cf066593aeb